test/model_simulation/simulate_ODEs.jl

```diff
@@ -1,30 +1,33 @@
-### Fetch Packages and Reaction Networks ###
+### Prepares Tests ###
 
 # Fetch packages.
-using Catalyst, OrdinaryDiffEq, Random, Test
-using ModelingToolkit: get_unknowns, get_ps
+using Catalyst, OrdinaryDiffEq, Test
 
-# Sets rnd number.
+# Sets stable rng number.
 using StableRNGs
 rng = StableRNG(12345)
 
-# Fetch test networks.
-@time include("../test_networks.jl")
+# Fetch test functions and networks.
+include("../test_functions.jl")
+include("../test_networks.jl")
 
 ### Compares to Known Solution ###
 
 # Exponential decay, should be identical to the (known) analytical solution.
 let
-    exponential_decay = @reaction_network begin d, X → ∅ end
+    exponential_decay = @reaction_network begin 
+        d, X → ∅
+    end
 
     for factor in [1e-2, 1e-1, 1e0, 1e1, 1e2]
-        u0 = factor * rand(rng, length(get_unknowns(exponential_decay)))
-        p = factor * rand(rng, length(get_ps(exponential_decay)))
-        prob = ODEProblem(exponential_decay, u0, (0.0, 100 / factor), p)
-        sol = solve(prob, Rosenbrock23(), saveat = range(0.0, 100 / factor, length = 101))
-        analytic_sol = map(t -> u0[1] * exp(-p[1] * t),
-                           range(0.0, 100 / factor, length = 101))
-        @test all(abs.(first.(sol.u) .- analytic_sol) .< 0.1)
+        u0 = rnd_u0(exponential_decay, rng; factor)    
+        t_stops = range(0.0, 100 / factor, length = 101)
+        p = rnd_ps(exponential_decay, rng; factor)
+        prob = ODEProblem(exponential_decay, u0, (0.0, t_stops[end]), p)
+
+        sol = solve(prob, Rosenbrock23(), saveat = t_stops, abstol = 1e-14, reltol = 1e-14)
+        analytic_sol = [u0[1][2] * exp(-p[1][2] * t) for t in t_stops]
+        @test sol[:X] ≈ analytic_sol
```

Comment:
Lots of simulation tests are basically makred as entirely new. The process is the same, but typically most lines had some change to accommodate new approach to maps etc.